game/scripts/vscripts/Sections/Chapter5/SectionRoshan.ts

```diff
@@ -0,0 +1,270 @@
+    sunsFanLocation: Vector(-5500, -4170, 256),
+    heroLocation: runeSpawnsLocations.topPowerUpRunePos.__add(Vector(-200, 0, -48)),
+    heroLocationTolerance: 800,
+    heroHasDoubleDamage: true,
+    blockades: [
+        chapter5Blockades.direJungleLowgroundRiver,
+        chapter5Blockades.topLaneRiver,
+        chapter5Blockades.radiantSecretShopRiver,
+        chapter5Blockades.direOutpostRiver,
+        chapter5Blockades.radiantAncientsRiver,
+        chapter5Blockades.radiantMidTopRiver,
+        chapter5Blockades.direMidTopRiver,
+        chapter5Blockades.midRiverTopSide,
+    ],
+};
+
+// Move to state when implementing final section of CH5
+const friendlyHeroesInfo = [
+    { name: "npc_dota_hero_tidehunter", loc: Vector(-2192, 1718, 0) }, // Offlane, tank
+    { name: "npc_dota_hero_juggernaut", loc: Vector(-2026, 1550, 0) }, // Carry
+    { name: "npc_dota_hero_mirana", loc: Vector(-2074, 1465, 0) }, // Position 4
+    { name: "npc_dota_hero_lion", loc: Vector(-2282, 1462, 0) }, // Position 5
+];
+
+const roshanMusic = "valve_ti10.music.roshan"
+
+function onStart(complete: () => void) {
+    print("Starting", sectionName);
+    const goalTracker = new GoalTracker();
+    const goalEnterRoshPit = goalTracker.addBoolean("Walk into Roshan's pit.");
+    const goalUpgradeTalents = goalTracker.addBoolean("Choose and pick between the available talents.")
+    const goalDefeatRoshan = goalTracker.addBoolean("Defeat Roshan!");
+    const goalPickupAegis = goalTracker.addBoolean("Pick up the Aegis of the Immortal.")
+    const goalLeaveRoshPit = goalTracker.addBoolean("Leave Roshan's pit and move to the next marker.")
+
+    const roshPitGoalPosition = Vector(-2600, 2200, 28)
+    const leaveRoshPitGoalPosition = Vector(-2140, 1740, 0)
+
+    const playerHero = getOrError(getPlayerHero())
+
+    const itemDaedalus = "item_greater_crit"
+    const itemAC = "item_assault"
+    const itemPowerTreads = "item_power_treads"
+    const itemHeart = "item_heart"
+    const itemAegis = "item_aegis"
+
+    let roshan = Entities.FindAllByName("npc_dota_roshan")[0] as CDOTA_BaseNPC
+
+    if (!roshan) {
+        roshan = CreateUnitByName("npc_dota_roshan", Vector(-2919, 2315, 32), true, undefined, undefined, DotaTeam.NEUTRALS)
+        roshan.FaceTowards(leaveRoshPitGoalPosition)
+        roshan.AddItemByName(itemAegis)
+    }
+
+    setupRoshanModifiers(roshan)
+
+    // Clear any Aegis boxes left on the ground
+    let droppedItems = Entities.FindAllByClassname("dota_item_drop") as CDOTA_Item_Physical[]
+
+    if (droppedItems) {
+        for (const droppedItem of droppedItems) {
+            let itemEntity = droppedItem.GetContainedItem()
+            if (itemEntity.GetAbilityName() === itemAegis) {
+                droppedItem.Destroy()
+            }
+        }
+    }
+
+    // DK lvl 25 talents
+    const dragonBlood25Talent = playerHero.FindAbilityByName("special_bonus_unique_dragon_knight") as CDOTABaseAbility
```

Comment:
This `as CDOTABaseAbility` cast is a bit ugly, nicest would be removing the cast and checking if the talents are not undefined, or if you are sure they are there just put a ! after the value to indicate it is definitely **never** undefined: 
`const dragonBlood25Talent = playerHero.FindAbilityByName("special_bonus_unique_dragon_knight");`

Also you're missing a bunch of ;'s

Reply:
I'll use the ! method, since if the player hero exists, so should the abilities for sure - and we use the getOrError with the call for getting the hero.

Do you mean the line ending ;'s? I saw it was barely used in this project from the start (though it seems to be more frequent now) so I kept to it as a convention, I think it's OK?

Reply:
I personally don't like using `;` but for this project I just kept using `;` in files that already used it. Imo as long as we don't mix it it's okay.